### PR TITLE
Add linking of Decisions to Amendments/Recons

### PR DESCRIPTION
**Pull request by dhaselhan**
* Combine Decision/Reconsideration/Amendment into their own Decision Module
* Add new boolean to track if outcome is for first or subsequent decisions
* Add new enable/disabling of controls to Form based on requirements

### CODE REVIEW

File: alcs-api/src/common/automapper/application-decision.automapper.profile.ts [hunk 1]
```diff
@@ -36,12 +36,38 @@ export class ApplicationDecisionProfile extends AutomapperProfile {
...
+        ),
+        forMember(
+          (a) => a.amends,
+          mapFrom((dec) =>
+            dec.amends
+              ? {
+                  uuid: dec.amends.uuid,
+                  linkedResolutions: dec.amends.amendsDecisions.map(
+                    (decision) =>
+                      `#${decision.resolutionNumber}/${dec.resolutionYear}`,
```
Comment (mhuseinov) [inline alcs-api/src/common/automapper/application-decision.automapper.profile.ts:61]: 👍

File: alcs-api/src/decision/application-decision/application-decision-outcome.entity.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Column, Entity } from 'typeorm';
+import { BaseCodeEntity } from '../../common/entities/base.code.entity';
+
+@Entity()
+export class DecisionOutcomeCode extends BaseCodeEntity {
+  @Column({
+    default: true,
+  })
+  forFirstDecision: boolean;
```
Comment (mhuseinov) [inline alcs-api/src/decision/application-decision/application-decision-outcome.entity.ts:9]: maybe isFirstDecision?

File: alcs-api/src/decision/application-decision/application-decision.controller.ts [hunk 3]
```diff
@@ -116,11 +137,35 @@ export class ApplicationDecisionController {
   @UserRoles(...ANY_AUTH_ROLE)
   async update(
     @Param('uuid') uuid: string,
-    @Body() appDecMeeting: UpdateApplicationDecisionDto,
+    @Body() updateDto: UpdateApplicationDecisionDto,
   ): Promise<ApplicationDecisionDto> {
+    if (updateDto.amendsUuid && updateDto.reconsidersUuid) {
+      throw new BadRequestException(
+        'Cannot set both an amends and reconsiders',
```
Comment (mhuseinov) [inline alcs-api/src/decision/application-decision/application-decision.controller.ts:144]: Is it a correct wording 'amends and reconsiders'?

File: alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts [hunk 4]
```diff
@@ -158,11 +214,15 @@ export class DecisionDialogComponent implements OnInit {
   }
 
   onSelectDecisionMaker(decisionMaker: DecisionMakerDto) {
-    if (decisionMaker.code !== 'CEOP') {
+    if (decisionMaker.code === 'CEOP') {
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts:217]: lets use enum for CEOP

File: alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts [hunk 5]
```diff
@@ -173,4 +233,23 @@ export class DecisionDialogComponent implements OnInit {
       });
     }
   }
+
+  onSelectPostDecision(postDecision: { type: string }) {
+    if (postDecision.type === 'amendment') {
+      this.form.controls.ceoCriterion.disable();
+      this.form.controls.outcome.disable();
+      this.form.controls.decisionMaker.disable();
+      this.ceoCriterion = this.data.ceoCriterion.filter((ceoCriterion) => ceoCriterion.code === 'MODI');
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts:242]: same about enums

File: alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts (hunk 4 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/application/decision/decision-dialog/decision-dialog.component.ts:217]: Done! Added an enum for Decision Maker

File: alcs-api/src/decision/application-decision/application-decision.controller.ts (hunk 3 above)
Comment (dhaselhan) [reply alcs-api/src/decision/application-decision/application-decision.controller.ts:144]: Updated to 'Cannot create a Decision linked to both an amendment and a reconsideration'

File: alcs-api/src/decision/application-decision/application-decision-outcome.entity.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-api/src/decision/application-decision/application-decision-outcome.entity.ts]: Done! Renamed and updated Migration.

File: alcs-api/src/decision/application-decision/application-decision.dto.ts [hunk 6]
```diff
@@ -42,6 +46,14 @@ export class UpdateApplicationDecisionDto {
...
   @IsOptional()
   chairReviewOutcome?: boolean | null;
+
+  @IsUUID()
+  @IsOptional()
+  amendsUuid?: string | null;
+
+  @IsUUID()
+  @IsOptional()
+  reconsidersUuid?: string | null;
```
Comment (mhuseinov) [inline alcs-api/src/decision/application-decision/application-decision.dto.ts:56]: just a nitpick, should this also be amendments and reconsiderations for consistency?

**Review by mhuseinov — APPROVED** [review]

**Pull request merged**